conans/paths/package_layouts/package_editable_layout.py

```diff
@@ -11,11 +11,12 @@
 
 class PackageEditableLayout(object):
 
-    def __init__(self, base_folder, layout_file, ref):
+    def __init__(self, base_folder, layout_file, ref, conanfile_path):
```

Comment:
Why I want to open this door, having a different path for the `conanfile` and the `base_folder`: when we have an out-of-sources recipe the source folder, build folder,... cannot be computed relative to the `conanfile.py` itself. Paths in the layout file should be absolute or we need to provide an additional path as _base folder_ (as it is requested here: https://github.com/conan-io/conan/issues/6896).